code/ship/ship.cpp

```diff
@@ -15808,7 +16466,7 @@ void ship_page_in()
 				for (j = 0; j < sip->n_subsystems; j++)
 					sip->subsystems[j].model_num = -1;
 
-				ship_copy_subsystem_fixup(sip);
+				ship_copy_subsystem_fixup(&(*sip));
```

Comment:
`ship_copy_subsystem_fixup()` takes a pointer. `sip` is now an iterator. To get a pointer to the `ship_info` object that the iterator points to, we need to deference the iterator and then get the address of that object. This ugly piece of code is the result. If there's a cleaner way to do it, I don't know one, and `making ship_copy_subsystem_fixup()` take an iterator instead of a pointer would require changing the other places it gets called to use iterators instead of pointers (which would be even more hackish).